narwhal/node/src/benchmark_client.rs

```diff
@@ -53,6 +59,68 @@ struct App {
     client_metric_port: u16,
 }
 
+/// A Narwhal client that instantiates LocalNarwhalClient lazily. (taken from consensus adapter)
+pub struct LazyNarwhalClient {
```

Comment:
I hope this code can be reused in some way. e.g. maybe we can move the code to worker instead.